system/src/system_sleep.cpp

```diff
@@ -45,45 +45,67 @@ using namespace particle;
 #undef LOG_COMPILE_TIME_LEVEL
 #define LOG_COMPILE_TIME_LEVEL LOG_LEVEL_ALL
 
-static bool system_sleep_network_suspend(network_interface_index index) {
-    bool resume = false;
+namespace {
+
+typedef struct {
+    bool suspended;
+    bool on;
+    bool connected;
+} network_status_t;
+
+network_status_t system_sleep_network_suspend(network_interface_index index) {
+    network_status_t status = {};
+    status.suspended = true;
+
     // Disconnect from network
     if (network_connecting(index, 0, NULL) || network_ready(index, 0, NULL)) {
         if (network_connecting(index, 0, NULL)) {
             network_connect_cancel(index, 1, 0, 0);
         }
         network_disconnect(index, NETWORK_DISCONNECT_REASON_SLEEP, NULL);
-        resume = true;
+        status.connected = true;
     }
 #if PLATFORM_GEN == 2
     if (!SPARK_WLAN_SLEEP) {
-        resume = true;
+        status.connected = true;
     }
 #endif
     // Turn off the modem
-    network_off(index, 0, 0, NULL);
-    LOG(TRACE, "Waiting interface %d to be off...", (int)index);
-    // There might be up to 30s delay to turn off the modem for particular platforms.
-    network_wait_off(index, 120000/*ms*/, nullptr);
-    return resume;
+    if (!network_is_off(index, nullptr)) {
+        status.on = true;
+        network_off(index, 0, 0, NULL);
+        LOG(TRACE, "Waiting interface %d to be off...", (int)index);
+        // There might be up to 30s delay to turn off the modem for particular platforms.
+        network_wait_off(index, 120000/*ms*/, nullptr);
+    } else {
+        LOG(TRACE, "Interface %d is off already", (int)index);
+    }
+    
+    return status;
 }
 
-static int system_sleep_network_resume(network_interface_index index) {
+int system_sleep_network_resume(network_interface_index index, network_status_t& status) {
```

Comment:
There is no need to pass `network_status_t` by reference here, which semantically means that it may be modified. Either by value (preferred) or at least const reference.